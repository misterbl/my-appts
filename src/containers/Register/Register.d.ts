import { FormikProps } from 'formik';
import { RouteComponentProps, StaticContext } from 'react-router';
import { InjectedIntlProps } from 'react-intl';
import { ActionCreatorsMapObject } from 'redux';
import { IAppState } from '../../types/state';

export interface IRegisterFormData {
  email?: string;
  password?: string;
}

export interface IRegisterFormDispatchToProps {
  apiThunk: ActionCreatorsMapObject;
}
export interface IRegisterFormMapStateToProps {
  user: IAppState['api']['userData'];
}

export interface IRegisterState {
  userExists: boolean;
}

export type IRegisterComponent = FormikProps<IRegisterFormData> &
  IRegisterFormDispatchToProps &
  IRegisterFormMapStateToProps &
  InjectedIntlProps &
  RouteComponentProps<any, StaticContext>;
