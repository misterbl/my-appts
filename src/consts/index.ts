import ROUTES from './routes';
import QUERIES from './queries';

export { ROUTES, QUERIES }