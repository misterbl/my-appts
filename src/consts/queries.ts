const queries = (args: any) => ({
  ADD_USER: `mutation {
    addUser (email: "${args.email}", avatar:"${args.avatar}" ) {
      _id
      firstName
      lastName
      avatar
      email
      address
      profileTitle
      profileDescription
     
    }
  }`,
  GET_USER: `mutation {
    getUser (email: "${args.email}") {
      _id
      firstName
      lastName
      avatar
      email
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
      unavailability
    }
  }`,
  GET_USER_BY_ID: `mutation {
    getUserById (_id: "${args.id}") {
      _id
      firstName
      lastName
      avatar
      email
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
    }
  }`,
  GET_USER_INFO: `mutation {
    getUser (email: "${args.email}") {
      _id
      firstName
      lastName
      avatar
      email
      address
    }
  }`,
  GET_AD_INFO: `mutation {
    getUser (email: "${args.email}") {
      _id
      profileTitle
      profileDescription
    }
  }`,
  UPDATE_AD_INFO: `mutation {
    updateUser (_id:"${args._id}" profileTitle: "${
    args.profileTitle
  }", profileDescription: "${args.profileDescription}") {
      _id
      firstName
      lastName
      avatar
      address
      profileTitle
      profileDescription
      children{name, dob, information, school, gender} 
      unavailability
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
    }
  }`,
  UPDATE_PERSONAL_INFO: `mutation {
    updateUser (_id: "${args._id}", firstName: "${
    args.firstName
  }", lastName: "${args.lastName}", address: "${args.address}" ) {
      _id
      firstName
      lastName
      avatar
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{name, dob, information, school, gender}
      unavailability
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}

    }
  }`,
  UPDATE_AVATAR: `mutation {
    updateUser (_id: "${args._id}", avatar:"${args.avatar}" ) {
      _id
      firstName
      lastName
      avatar
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}

    }
  }`,
  UPDATE_LOCATION: `mutation {
    updateUser (_id: "${args._id}", address: "${args.address}", lat: "${
    args.lat
  }", lng: "${args.lng}" ) {
      _id
      firstName
      lastName
      avatar
      address
      lat
      lng
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{name, dob, information, school, gender}
      unavailability
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}

    }
  }`,

  UPDATE_ABOUT_YOU: `mutation {
    updateUser (_id: "${args._id}", drivingLicense: ${
    args.drivingLicense
  }, nonSmoker: ${args.nonSmoker}, car:${args.car}) {
      _id
      firstName
      lastName
      avatar
      address
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      children{name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
      unavailability

    }
  }`,
  UPDATE_AVAILABILITY: `mutation {
    updateUser (_id: "${args._id}", availability: ${JSON.stringify(
    JSON.stringify(args.availability),
  )}) {
      _id
      firstName
      lastName
      avatar
      address
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      unavailability
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
    }
  }`,
  UPDATE_UNAVAILABILITY: `mutation {
    updateUser (_id: "${args._id}", unavailability: ${JSON.stringify(
    args.unavailability,
  )}) {
      _id
      firstName
      lastName
      avatar
      address
      drivingLicense
      nonSmoker
      car
      profileTitle
      profileDescription
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
      unavailability
    }
  }`,
  ADD_CHILD: `mutation {
    updateUser (_id: "${args._id}", child: ${JSON.stringify(
    JSON.stringify(args.child),
  )}
) {
      _id
      firstName
      lastName
      avatar
      address
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
    }
  }`,
  UPDATE_CHILD: `mutation {
    updateChild (_id: "${args._id}", child: ${JSON.stringify(
    JSON.stringify(args.child),
  )}
) {
      _id
      firstName
      lastName
      avatar
      address
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
      availability{beforeSchool, morning, noon, afternoon, afterSchool, evening, night}
      unavailability
    }
  }`,
  REMOVE_CHILD: `mutation {
    removeChild (_id: "${args._id}", child: ${JSON.stringify(
    JSON.stringify(args.child),
  )}
) {
      _id
      firstName
      lastName
      avatar
      address
      profileTitle
      profileDescription
      children{id, name, dob, information, school, gender}
    }
  }`,
});

export default queries;
