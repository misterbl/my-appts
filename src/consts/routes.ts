const routes = {
  INDEX: '/',
  CARD: '/card',
  SIGN_IN: '/signin',
  REGISTER: '/register',
  DASHBOARD: './dashboard',
  INBOX: './inbox',
  FAVOURITE: './favourite',
  SEARCH: './search',
};

export default routes;
