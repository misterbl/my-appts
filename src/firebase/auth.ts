import { auth, provider } from './firebase';

// Sign Up
export const doCreateUserWithEmailAndPassword = (email: string, password: string) =>
  auth.createUserWithEmailAndPassword(email, password);

  // Sign In
export const doSignInWithEmailAndPassword = (email: string, password: string) =>
auth.signInWithEmailAndPassword(email, password);

// Sign out
export const doSignOut = () =>
  auth.signOut();


  // Password Reset
export const doPasswordReset = (email: string) =>
auth.sendPasswordResetEmail(email);

// Password Change
export const doPasswordUpdate = (password: string) =>
auth && auth.currentUser && auth.currentUser.updatePassword(password);

// Facebook
export const doFacebookSignIn = () => auth.signInWithPopup(provider).then(function(result) {
    // This gives you a Facebook Access Token. You can use it to access the Facebook API.
    const token = result;
    // The signed-in user info.
    const user = result.user;
    console.log("user", user);
    console.log("token", token);
    
    
    // ...
  }).catch(function(error) {
    // Handle Errors here.
    const errorCode = error.code;
    const errorMessage = error.message;
    console.log("errorCode", errorCode);
    console.log("errorMessage", errorMessage);
    
    // The email of the user's account used.
    const email = error.email;
    console.log("email", email);
    
    // The firebase.auth.AuthCredential type that was used.
    const credential = error.credential;
    console.log("credential", credential);
    
    // ...
  });