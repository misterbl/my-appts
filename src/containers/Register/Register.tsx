import * as React from 'react';
import { Formik, Form } from 'formik';
import { auth } from '../../firebase';
import { injectIntl, FormattedMessage } from 'react-intl';
import { withRouter } from 'react-router-dom';
import { IRegisterComponent } from './Register.d';
import ROUTES from '../../consts/routes';

export class Register extends React.Component<IRegisterComponent> {
  onSubmit = (event: any) => {
    const { email, passwordOne } = event;
    auth
      .doCreateUserWithEmailAndPassword(email, passwordOne)
      .then(authUser => {
        this.props.history.push(ROUTES.DASHBOARD);
      })
      .catch(error => {
        console.log(error);
      });
  };

  faceBookLogin = () => {
    auth.doFacebookSignIn();
    this.props.history.push(ROUTES.DASHBOARD);
  };

  signOut = () => {
    auth.doSignOut();
  };
  render() {
    const { formatMessage } = this.props.intl;

    return (
      <div className="flex flex-column vh-100">
        <button onClick={this.signOut}>Logout</button>
        <button onClick={this.faceBookLogin} className="facebook-button" />
        <span className="white tc mt2">
          <FormattedMessage id="general|or" />
        </span>

        <Formik
          initialValues={{
            email: '',
            passwordOne: '',
            passwordTwo: '',
          }}
          onSubmit={this.onSubmit}
        >
          {({ values, isSubmitting, setFieldValue }) => (
            <Form className="signin-register flex flex-column">
              <label htmlFor="email" />
              <input
                value={values.email}
                name="email"
                onChange={event => setFieldValue('email', event.target.value)}
                type="text"
                placeholder={formatMessage({
                  id: 'general|placeholder|email',
                })}
              />
              <label htmlFor="passwordOne" />
              <input
                value={values.passwordOne}
                name="passwordOne"
                onChange={event =>
                  setFieldValue('passwordOne', event.target.value)
                }
                type="password"
                placeholder={formatMessage({
                  id: 'general|placeholder|password',
                })}
              />
              <label htmlFor="passwordTwo" />
              <input
                value={values.passwordTwo}
                name="passwordTwo"
                onChange={event =>
                  setFieldValue('passwordTwo', event.target.value)
                }
                type="password"
                placeholder={formatMessage({
                  id: 'general|placeholder|confirmpassword',
                })}
              />
              <button
                className="bg-green white fw7 ph3 ttc di pv3 bn-ns"
                type="submit"
                disabled={isSubmitting}
              >
                <FormattedMessage id="general|button|next" />
              </button>

              {/* {error && <p>{error.message}</p>} */}
            </Form>
          )}
        </Formik>
      </div>
    );
  }
}

const injectIntlRegister = injectIntl(Register);
export default withRouter(injectIntlRegister);
