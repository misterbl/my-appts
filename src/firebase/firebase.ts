import * as firebase from 'firebase/app';
import 'firebase/auth';


const config = {
    apiKey: "AIzaSyAUFtb8M2Pzfgpn48-F736NVgKlCcc2haQ",
    authDomain: "kidappi-app.firebaseapp.com",
    databaseURL: "https://kidappi-app.firebaseio.com",
    projectId: "kidappi-app",
    storageBucket: "kidappi-app.appspot.com",
    messagingSenderId: "600523020740"
  };

  if (!firebase.apps.length) {
    firebase.initializeApp(config);
  }

  const auth = firebase.auth();
  auth.useDeviceLanguage();
  const provider = new firebase.auth.FacebookAuthProvider()
export {
  auth,
  provider,
};