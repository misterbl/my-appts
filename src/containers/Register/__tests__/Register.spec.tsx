import * as React from 'react';
import { shallow } from 'enzyme';
import formikRenderMock from '../../../testMocks/formikRender.mock';
import historyMock from '../../../testMocks/history.mock';
import { Register, RegisterForm } from '../Register';

import { auth } from '../../../firebase';
// import * as firebase from 'firebase/app';
import { ROUTES } from '../../../consts';
import IntlMock from '../../../testMocks/intl.mock';
import generateAppState from '../../../testMocks/appState.mock';

jest.mock('../../../firebase', () => ({
  auth: {
    doCreateUserWithEmailAndPassword: jest
      .fn()
      .mockReturnValue(Promise.resolve({})),
    doFacebookSignIn: jest.fn().mockReturnValue(Promise.resolve({})),
  },
}));

describe('Register', () => {
  const props = {
    values: {
      email: 'emailTest',
      password: 'passwordTest',
    },
    // errors: { email: null, password: null },
    // touched: { email: false, password: false },
    ...formikRenderMock,
    initialValues: {
      email: '',
      password: '',
    },
    apiThunk: { getUserData: jest.fn(), postUserData: jest.fn() },
    intl: IntlMock,
    history: historyMock,
    location: historyMock.location,
    match: {
      params: {},
      isExact: true,
      path: '',
      url: '',
    },
    onSubmit: jest.fn(),
    faceBookRegister: jest.fn(),
  };
  // @ts-ignore
  const wrapper = shallow(<Register {...props} />);
  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a Facebook element', () => {
    expect(wrapper.find('FacebookButton').length).toBe(1);
  });
  describe('Facebook button', () => {
    const facebookButton = wrapper.find('FacebookButton');
    it('calls doFacebookSignIn on Facebook button click', async () => {
      await facebookButton.simulate('click');
      expect(auth.doFacebookSignIn).toHaveBeenCalled();
    });
    it('calls getUserData on Facebook button click', async () => {
      await facebookButton.simulate('click');
      expect(props.apiThunk.getUserData).toHaveBeenCalled();
    });
    it('calls history.push on Facebook button click', async () => {
      await facebookButton.simulate('click');
      expect(props.history.push).toHaveBeenCalledWith(ROUTES.DASHBOARD);
    });
  });

  it('should render a Formik element', () => {
    expect(wrapper.find('Formik').length).toBe(1);
  });

  it('should call doCreateUserWithEmailAndPassword on form submit', () => {
    const {
      values: { email, password },
    } = props;
    const formik = wrapper.find('Formik');
    formik.simulate('submit', { email, password });
    expect(auth.doCreateUserWithEmailAndPassword).toHaveBeenCalled();
  });
  describe('Errors', () => {
    const newProps = {
      ...props,
      userExists: false,
      user: generateAppState().api.userData,
    };
    const registerForm = shallow(<RegisterForm {...newProps} />);
    it('it shows an error message for email when there is an error', () => {
      registerForm.setProps({
        errors: { email: 'emailError' },
        touched: { email: true },
      });
      const errroMessage = registerForm.find('ErrorMessage');
      expect(errroMessage.length).toBe(1);
    });
    it('it shows an error message for password when there is an error', () => {
      registerForm.setProps({
        errors: { password: 'passwordError' },
        touched: { password: true },
      });
      const errroMessage = registerForm.find('ErrorMessage');
      expect(errroMessage.length).toBe(1);
    });
    it('it shows an error message for userExists when there is an error', () => {
      registerForm.setProps({
        errors: {},
        touched: {},
        userExists: true,
      });
      const errroMessage = registerForm.find('ErrorMessage');
      expect(errroMessage.length).toBe(1);
    });
  });
});
