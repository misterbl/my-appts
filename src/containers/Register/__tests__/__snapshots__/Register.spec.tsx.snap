// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    apiThunk={
        Object {
            "getUserData": [Function],
            "postUserData": [Function],
          }
    }
    dirty={false}
    errors={Object {}}
    faceBookRegister={[Function]}
    handleBlur={[Function]}
    handleChange={[Function]}
    handleReset={[Function]}
    handleSubmit={[Function]}
    history={
        Object {
            "action": "PUSH",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 123,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "",
              "pathname": "",
              "search": "",
              "state": "",
            },
            "push": [Function],
            "replace": [Function],
          }
    }
    initialValues={
        Object {
            "email": "",
            "password": "",
          }
    }
    intl={
        Object {
            "defaultFormats": "",
            "defaultLocale": "",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": "",
            "locale": "",
            "messages": Object {},
            "now": [Function],
          }
    }
    isSubmitting={false}
    isValid={false}
    isValidating={false}
    location={
        Object {
            "hash": "",
            "key": "",
            "pathname": "",
            "search": "",
            "state": "",
          }
    }
    match={
        Object {
            "isExact": true,
            "params": Object {},
            "path": "",
            "url": "",
          }
    }
    onSubmit={[Function]}
    registerField={[Function]}
    resetForm={[Function]}
    setError={[Function]}
    setErrors={[Function]}
    setFieldError={[Function]}
    setFieldTouched={[Function]}
    setFieldValue={[Function]}
    setFormikState={[Function]}
    setStatus={[Function]}
    setSubmitting={[Function]}
    setTouched={[Function]}
    setValues={[Function]}
    submitCount={0}
    submitForm={[Function]}
    touched={Object {}}
    unregisterField={[Function]}
    validateField={[Function]}
    validateForm={[Function]}
    values={
        Object {
            "email": "emailTest",
            "password": "passwordTest",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          className="kidappi white mt4 tc w-100"
>
          Kidappi
</a>,
        <FacebookButton
          className="mt4 shadow-5"
          onClick={[Function]}
/>,
        <div
          className="text--line-through"
>
          <span
                    className="spant3 ma0 tc white text--line-through--text"
          >
                    <FormattedMessage
                              id="general|or"
                              values={Object {}}
                    />
          </span>
</div>,
        <p
          className="tc white f4"
>
          <FormattedMessage
                    id="content|regiter|registerwithemail"
                    values={Object {}}
          />
</p>,
        <Formik
          enableReinitialize={false}
          initialValues={
                    Object {
                              "email": "",
                              "password": "",
                            }
          }
          isInitialValid={false}
          onSubmit={[Function]}
          render={[Function]}
          validateOnBlur={true}
          validateOnChange={true}
          validationSchema={
                    ObjectSchema {
                              "_blacklist": RefSet {
                                "list": Set {},
                                "refs": Map {},
                              },
                              "_conditions": Array [],
                              "_defaultDefault": [Function],
                              "_deps": Array [],
                              "_excludedEdges": Array [],
                              "_exclusive": Object {},
                              "_mutate": false,
                              "_nodes": Array [
                                "password",
                                "email",
                              ],
                              "_options": Object {
                                "abortEarly": true,
                                "recursive": true,
                              },
                              "_type": "object",
                              "_typeError": [Function],
                              "_whitelist": RefSet {
                                "list": Set {},
                                "refs": Map {},
                              },
                              "fields": Object {
                                "email": StringSchema {
                                  "_blacklist": RefSet {
                                    "list": Set {},
                                    "refs": Map {},
                                  },
                                  "_conditions": Array [],
                                  "_deps": Array [],
                                  "_exclusive": Object {
                                    "required": false,
                                    "undefined": false,
                                  },
                                  "_mutate": false,
                                  "_options": Object {
                                    "abortEarly": true,
                                    "recursive": true,
                                  },
                                  "_type": "string",
                                  "_typeError": [Function],
                                  "_whitelist": RefSet {
                                    "list": Set {},
                                    "refs": Map {},
                                  },
                                  "tests": Array [
                                    [Function],
                                    [Function],
                                    [Function],
                                  ],
                                  "transforms": Array [
                                    [Function],
                                  ],
                                },
                                "password": StringSchema {
                                  "_blacklist": RefSet {
                                    "list": Set {},
                                    "refs": Map {},
                                  },
                                  "_conditions": Array [],
                                  "_deps": Array [],
                                  "_exclusive": Object {
                                    "min": true,
                                    "required": false,
                                  },
                                  "_mutate": false,
                                  "_options": Object {
                                    "abortEarly": true,
                                    "recursive": true,
                                  },
                                  "_type": "string",
                                  "_typeError": [Function],
                                  "_whitelist": RefSet {
                                    "list": Set {},
                                    "refs": Map {},
                                  },
                                  "tests": Array [
                                    [Function],
                                    [Function],
                                    [Function],
                                  ],
                                  "transforms": Array [
                                    [Function],
                                  ],
                                },
                              },
                              "tests": Array [],
                              "transforms": Array [
                                [Function],
                              ],
                            }
          }
/>,
      ],
      "className": "flex flex-column vh-100 green-bg ph4 ph7-l ph6-m pt5-ns",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Kidappi",
          "className": "kidappi white mt4 tc w-100",
        },
        "ref": null,
        "rendered": "Kidappi",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "mt4 shadow-5",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="spant3 ma0 tc white text--line-through--text"
>
            <FormattedMessage
                        id="general|or"
                        values={Object {}}
            />
</span>,
          "className": "text--line-through",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormattedMessage
              id="general|or"
              values={Object {}}
/>,
            "className": "spant3 ma0 tc white text--line-through--text",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "general|or",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FormattedMessage
            id="content|regiter|registerwithemail"
            values={Object {}}
/>,
          "className": "tc white f4",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "content|regiter|registerwithemail",
            "values": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "enableReinitialize": false,
          "initialValues": Object {
            "email": "",
            "password": "",
          },
          "isInitialValid": false,
          "onSubmit": [Function],
          "render": [Function],
          "validateOnBlur": true,
          "validateOnChange": true,
          "validationSchema": ObjectSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_defaultDefault": [Function],
            "_deps": Array [],
            "_excludedEdges": Array [],
            "_exclusive": Object {},
            "_mutate": false,
            "_nodes": Array [
              "password",
              "email",
            ],
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "object",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "fields": Object {
              "email": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "required": false,
                  "undefined": false,
                },
                "_mutate": false,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                  [Function],
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
              "password": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "min": true,
                  "required": false,
                },
                "_mutate": false,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                  [Function],
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
            },
            "tests": Array [],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="kidappi white mt4 tc w-100"
>
            Kidappi
</a>,
          <FacebookButton
            className="mt4 shadow-5"
            onClick={[Function]}
/>,
          <div
            className="text--line-through"
>
            <span
                        className="spant3 ma0 tc white text--line-through--text"
            >
                        <FormattedMessage
                                    id="general|or"
                                    values={Object {}}
                        />
            </span>
</div>,
          <p
            className="tc white f4"
>
            <FormattedMessage
                        id="content|regiter|registerwithemail"
                        values={Object {}}
            />
</p>,
          <Formik
            enableReinitialize={false}
            initialValues={
                        Object {
                                    "email": "",
                                    "password": "",
                                  }
            }
            isInitialValid={false}
            onSubmit={[Function]}
            render={[Function]}
            validateOnBlur={true}
            validateOnChange={true}
            validationSchema={
                        ObjectSchema {
                                    "_blacklist": RefSet {
                                      "list": Set {},
                                      "refs": Map {},
                                    },
                                    "_conditions": Array [],
                                    "_defaultDefault": [Function],
                                    "_deps": Array [],
                                    "_excludedEdges": Array [],
                                    "_exclusive": Object {},
                                    "_mutate": false,
                                    "_nodes": Array [
                                      "password",
                                      "email",
                                    ],
                                    "_options": Object {
                                      "abortEarly": true,
                                      "recursive": true,
                                    },
                                    "_type": "object",
                                    "_typeError": [Function],
                                    "_whitelist": RefSet {
                                      "list": Set {},
                                      "refs": Map {},
                                    },
                                    "fields": Object {
                                      "email": StringSchema {
                                        "_blacklist": RefSet {
                                          "list": Set {},
                                          "refs": Map {},
                                        },
                                        "_conditions": Array [],
                                        "_deps": Array [],
                                        "_exclusive": Object {
                                          "required": false,
                                          "undefined": false,
                                        },
                                        "_mutate": false,
                                        "_options": Object {
                                          "abortEarly": true,
                                          "recursive": true,
                                        },
                                        "_type": "string",
                                        "_typeError": [Function],
                                        "_whitelist": RefSet {
                                          "list": Set {},
                                          "refs": Map {},
                                        },
                                        "tests": Array [
                                          [Function],
                                          [Function],
                                          [Function],
                                        ],
                                        "transforms": Array [
                                          [Function],
                                        ],
                                      },
                                      "password": StringSchema {
                                        "_blacklist": RefSet {
                                          "list": Set {},
                                          "refs": Map {},
                                        },
                                        "_conditions": Array [],
                                        "_deps": Array [],
                                        "_exclusive": Object {
                                          "min": true,
                                          "required": false,
                                        },
                                        "_mutate": false,
                                        "_options": Object {
                                          "abortEarly": true,
                                          "recursive": true,
                                        },
                                        "_type": "string",
                                        "_typeError": [Function],
                                        "_whitelist": RefSet {
                                          "list": Set {},
                                          "refs": Map {},
                                        },
                                        "tests": Array [
                                          [Function],
                                          [Function],
                                          [Function],
                                        ],
                                        "transforms": Array [
                                          [Function],
                                        ],
                                      },
                                    },
                                    "tests": Array [],
                                    "transforms": Array [
                                      [Function],
                                    ],
                                  }
            }
/>,
        ],
        "className": "flex flex-column vh-100 green-bg ph4 ph7-l ph6-m pt5-ns",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Kidappi",
            "className": "kidappi white mt4 tc w-100",
          },
          "ref": null,
          "rendered": "Kidappi",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "mt4 shadow-5",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="spant3 ma0 tc white text--line-through--text"
>
              <FormattedMessage
                            id="general|or"
                            values={Object {}}
              />
</span>,
            "className": "text--line-through",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FormattedMessage
                id="general|or"
                values={Object {}}
/>,
              "className": "spant3 ma0 tc white text--line-through--text",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "general|or",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormattedMessage
              id="content|regiter|registerwithemail"
              values={Object {}}
/>,
            "className": "tc white f4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "content|regiter|registerwithemail",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "enableReinitialize": false,
            "initialValues": Object {
              "email": "",
              "password": "",
            },
            "isInitialValid": false,
            "onSubmit": [Function],
            "render": [Function],
            "validateOnBlur": true,
            "validateOnChange": true,
            "validationSchema": ObjectSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_defaultDefault": [Function],
              "_deps": Array [],
              "_excludedEdges": Array [],
              "_exclusive": Object {},
              "_mutate": false,
              "_nodes": Array [
                "password",
                "email",
              ],
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "object",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "fields": Object {
                "email": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "required": false,
                    "undefined": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "password": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "min": true,
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
              },
              "tests": Array [],
              "transforms": Array [
                [Function],
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
